workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -480,6 +506,24 @@ def compute_first_rows_response(
             raise FeaturesError("The split features (columns) cannot be extracted.", cause=err) from err
     else:
         features = info.features
+
+    if len(features) > columns_max_number:
+        raise TooManyColumnsError(
+            f"Too many columns. The maximum supported number of columns is {columns_max_number}."
+        )
+
+    # validate size of response without the rows
+    features_list = to_features_list(features=features)
+    response: FirstRowsResponse = {
+        "dataset": dataset,
+        "config": config,
+        "split": split,
+        "features": features_list,
+        "rows": [],
+    }
+
+    validate_content_size(response, rows_max_bytes=rows_max_bytes)
+
     # get the rows
     try:
         rows = get_rows(
```

Comment:
You should also check the number of columns here, in case `features` is `None`

Reply:
Yes, I moved that validation to line 510